From 01a6cd14c4bc6780511c905a1071b091496d3ce3 Mon Sep 17 00:00:00 2001
From: Numan Siddique <nusiddiq@redhat.com>
Date: Tue, 27 Dec 2016 16:38:49 +0530
Subject: [PATCH 2/2] ovn-ctl: Add the inactivity probe option

Signed-off-by: Numan Siddique <nusiddiq@redhat.com>
---
 ovn/utilities/ovn-ctl | 41 +++++++++++++++++++++++++++++++++--------
 1 file changed, 33 insertions(+), 8 deletions(-)

diff --git a/ovn/utilities/ovn-ctl b/ovn/utilities/ovn-ctl
index a696d5e..93aaf34 100755
--- a/ovn/utilities/ovn-ctl
+++ b/ovn/utilities/ovn-ctl
@@ -102,10 +102,6 @@ start_ovsdb () {
         set "$@" --certificate=db:OVN_Northbound,SSL,certificate
         set "$@" --ca-cert=db:OVN_Northbound,SSL,ca_cert
 
-        if test X"$DB_NB_CREATE_INSECURE_REMOTE" = Xyes; then
-            set "$@" --remote=ptcp:$DB_NB_PORT:$DB_NB_ADDR
-        fi
-
         if test ! -z "$DB_NB_SYNC_FROM_ADDR"; then
             echo "$DB_NB_SYNC_FROM_PROTO:$DB_NB_SYNC_FROM_ADDR:$DB_NB_SYNC_FROM_PORT" > $ovnnb_active_conf_file
         fi
@@ -116,6 +112,20 @@ start_ovsdb () {
 
         $@ $DB_NB_FILE
         ovn-nbctl init
+
+	if test X"$DB_NB_CREATE_INSECURE_REMOTE" = Xyes; then
+            conn_info=`ovn-nbctl  find Connection target="ptcp\:$DB_NB_PORT\:$DB_NB_ADDR"`
+            conn_uuid=`echo $conn_info | awk '{print $3'}`
+            if test X"$conn_uuid" = X; then
+                ovn-nbctl set-connection "ptcp:$DB_NB_PORT:$DB_NB_ADDR"
+                conn_info=`ovn-nbctl  find Connection target="ptcp\:$DB_NB_PORT\:$DB_NB_ADDR"`
+                conn_uuid=`echo $conn_info | awk '{print $3'}`
+            fi
+
+            if test X"$DB_NB_INACTIVITY_PROBE" != X; then
+                ovn-nbctl set Connection $conn_uuid inactivity_probe=$DB_NB_INACTIVITY_PROBE
+            fi
+        fi
     fi
 
     # Check and eventually start ovsdb-server for Southbound DB
@@ -133,10 +143,6 @@ start_ovsdb () {
         set "$@" --certificate=db:OVN_Southbound,SSL,certificate
         set "$@" --ca-cert=db:OVN_Southbound,SSL,ca_cert
 
-        if test X"$DB_NB_CREATE_INSECURE_REMOTE" = Xyes; then
-            set "$@" --remote=ptcp:$DB_SB_PORT:$DB_SB_ADDR
-        fi
-
         if test ! -z "$DB_SB_SYNC_FROM_ADDR"; then
             echo "$DB_SB_SYNC_FROM_PROTO:$DB_SB_SYNC_FROM_ADDR:$DB_SB_SYNC_FROM_PORT" > $ovnsb_active_conf_file
         fi
@@ -147,6 +153,20 @@ start_ovsdb () {
 
         $@ $DB_SB_FILE
         ovn-sbctl init
+
+	if test X"$DB_SB_CREATE_INSECURE_REMOTE" = Xyes; then
+            conn_info=`ovn-sbctl  find Connection target="ptcp\:$DB_SB_PORT\:$DB_SB_ADDR"`
+            conn_uuid=`echo $conn_info | awk '{print $3'}`
+            if test X"$conn_uuid" = X; then
+                ovn-sbctl set-connection "ptcp:$DB_SB_PORT:$DB_SB_ADDR"
+                conn_info=`ovn-sbctl  find Connection target="ptcp\:$DB_SB_PORT\:$DB_SB_ADDR"`
+                conn_uuid=`echo $conn_info | awk '{print $3'}`
+            fi
+
+            if test X"$DB_SB_INACTIVITY_PROBE" != X; then
+                ovn-sbctl set Connection $conn_uuid inactivity_probe=$DB_SB_INACTIVITY_PROBE
+            fi
+        fi
     fi
 }
 
@@ -338,6 +358,9 @@ set_defaults () {
 
     DB_SB_CREATE_INSECURE_REMOTE="no"
     DB_NB_CREATE_INSECURE_REMOTE="no"
+
+    DB_NB_INACTIVITY_PROBE=""
+    DB_SB_INACTIVITY_PROBE=""
 }
 
 set_option () {
@@ -413,10 +436,12 @@ File location options:
   --db-nb-sync-from-port=PORT OVN Northbound active db tcp port (default: $DB_NB_SYNC_FROM_PORT)
   --db-nb-sync-from-proto=PROTO OVN Northbound active db transport (default: $DB_NB_SYNC_FROM_PROTO)
   --db-nb-create-insecure-remote=yes|no Create ptcp OVN Northbound remote (default: $DB_NB_CREATE_INSECURE_REMOTE)
+  --db-nb-inactivity-probe=TIME Set inactivity probe (in msec) for NB remote (default:$DB_NB_INACTIVITY_PROBE)
   --db-sb-sync-from-addr=ADDR OVN Southbound active db tcp address (default: $DB_SB_SYNC_FROM_ADDR)
   --db-sb-sync-from-port=ADDR OVN Southbound active db tcp port (default: $DB_SB_SYNC_FROM_PORT)
   --db-sb-sync-from-proto=PROTO OVN Southbound active db transport (default: $DB_SB_SYNC_FROM_PROTO)
   --db-sb-create-insecure-remote=yes|no Create ptcp OVN Southbound remote (default: $DB_SB_CREATE_INSECURE_REMOTE)
+  --db-sb-inactivity-probe=TIME Set inactivity probe (in msec) for SB remote (default: $DB_SB_INACTIVITY_PROBE)
 
 Default directories with "configure" option and environment variable override:
   logs: /usr/local/var/log/openvswitch (--with-logdir, OVS_LOGDIR)
-- 
2.9.3

